src/main/java/com/puppycrawl/tools/checkstyle/filters/JavaPatchFilterElement.java

```diff
@@ -43,12 +45,43 @@
      */
     private static final String MAX = "max";
 
+    /**
+     * Map that mapping between a check and its ancestor token types.
+     */
+    private static final Map<String, List<Integer>> CHECK_TO_ANCESTOR_NODES_MAP =
+            new HashMap<String, List<Integer>>() {
+        {
+            put("ArrayTrailingComma", Arrays.asList(TokenTypes.METHOD_DEF));
+            put("AvoidNestedBlocks", Arrays.asList(TokenTypes.METHOD_DEF));
```

Comment:
what if nested block is not in a method?
I think SLIST node is what this Check needs to cover all cases. Please recheck.
Example AST: http://rveach.no-ip.org/checkstyle/checkstyle.php?action=view&config=72fba57f86eaf7df54f418b381c28422&code=1aff4d8fc71b843b8429b429e5cd64a8&checkstyle=checkstyle-8.35-all.jar&printTree=true&printSuppressions=

Reply:
@baratali this example can not be covered if we use SLIST node, so I think we still need use METHOD_DEF

```
-        if (x == 1)
         {  // violation without filter
             x = 2;
         }
```

```
|--LITERAL_IF -> if [18:8]
    |       |   |--LPAREN -> ( [18:11]
    |       |   |--EXPR -> EXPR [18:14]
    |       |   |   `--EQUAL -> == [18:14]
    |       |   |       |--IDENT -> x [18:12]
    |       |   |       `--NUM_INT -> 1 [18:17]
    |       |   |--RPAREN -> ) [18:18]
    |       |   `--SLIST -> { [19:8]
    |       |       |--EXPR -> EXPR [20:14]
    |       |       |   `--ASSIGN -> = [20:14]
    |       |       |       |--SINGLE_LINE_COMMENT -> // [19:11]
    |       |       |       |   `--COMMENT_CONTENT ->  violation without filter\n [19:13]
    |       |       |       |--IDENT -> x [20:12]
    |       |       |       `--NUM_INT -> 2 [20:16]
    |       |       |--SEMI -> ; [20:17]
    |       |       `--RCURLY -> } [21:8]
```

Reply:
If your example is in method for example, then SLIST will be a method's body. If it's in other block, then SLIST will be this block's body.
Example AST with a method: http://rveach.no-ip.org/checkstyle/checkstyle.php?action=view&config=72fba57f86eaf7df54f418b381c28422&code=f44894f4f1f5e0f263488ab32aa9c3f3&checkstyle=checkstyle-8.35-all.jar&printTree=true&printSuppressions=

Reply:
yes, it's ok. changed